dotCMS/src/main/java/com/dotcms/rest/api/v1/site/SiteResource.java

```diff
@@ -843,6 +843,117 @@ public Response createNewSite(@Context final HttpServletRequest httpServletReque
                 this.toView(this.siteHelper.save(newSite, user, pageMode.respectAnonPerms)))).build();
     }
 
+    /**
+     * Updates a site
+     * @param httpServletRequest
+     * @param httpServletResponse
+     * @param newSiteForm
+     * @return
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws PortalException
+     * @throws SystemException
+     * @throws ParseException
+     * @throws SchedulerException
+     * @throws ClassNotFoundException
+     */
+    @PUT
+    @JSONP
+    @NoCache
+    @Produces({MediaType.APPLICATION_JSON, "application/javascript"})
+    public Response updateSite(@Context final HttpServletRequest httpServletRequest,
+                                  @Context final HttpServletResponse httpServletResponse,
+                                  @QueryParam("id") final String  siteIdentifier,
+                                  final SiteForm newSiteForm)
+            throws DotDataException, DotSecurityException, PortalException, SystemException, ParseException, SchedulerException, ClassNotFoundException {
+
+        final User user = new WebResource.InitBuilder(this.webResource)
+                .requestAndResponse(httpServletRequest, httpServletResponse)
```

Comment:
I would add access to the `sites` portlet as a requirement to use this tooling?

Reply:
done. I also removed the license required from the copy site as requested by Stefan